src/System.Private.ServiceModel/src/System/ServiceModel/Channels/BodyWriter.cs

```diff
@@ -80,6 +81,10 @@ internal BodyWriter OnCreateBufferedCopy(int maxBufferSize, XmlDictionaryReaderQ
         }
 
         protected abstract void OnWriteBodyContents(XmlDictionaryWriter writer);
+        protected virtual Task OnWriteBodyContentsAsync(XmlDictionaryWriter writer)
+        {
+            throw new NotImplementedException();
```

Comment:
Recommend making @iamjasonp 's fixes and opening a new issue to consider updating the other derived BodyWriters to use the new virtual for async